h/views/oidc.py

```diff
@@ -81,10 +86,12 @@ def __init__(self, request: Request) -> None:
 
     @view_config(is_authenticated=True)
     def redirect(self):
+        expected_state = self._request.session.pop(ORCID_STATE_SESSION_KEY, None)
```

Comment:
The view now removes the `state` from the session itself and passes it to `OAuth2RedirectSchema.validate()` as `expected_state` below.